src/ts/app/editor/src/render-run-button.ts

```diff
@@ -0,0 +1,6 @@
+import {html, TemplateResult} from "lit-html"
+
+
+export function renderRunButton(userCanStartCodeRun: boolean): TemplateResult {
```

Comment:
hmmm, I'm not quite sure I get the difference between the root renderer and children. could you also set this up so renderRunButton was a subscription to the state tree...

If I were to re-work this, there would be a file with all of the HTML, i.e.

```
<div>
  <div id="code-window"></div>
  <div id="run-button"></div>
</div>
```

ANd then the render functions would subscribe to the state, be passed in their "base HTML" be able to find their nodes, and modify them. Reason is you could at least see the structure in HTML... need to play with this  format more, maybe this makes sense but I can't picture what the page is actually going to look like right now.

Reply:
I'm not quite sure about what the best overall organization of subscribers/renderers is - I think there are a couple of mildly-competing ideas about that in the codebase at present.

However I do have a principle that I try to mostly follow, about functions, which I think applies here:

* if a function can be stateless, it should be - i.e. it has no side effects
* if a function can have a narrower knowledge of input and output state, it should

^ wdyt about these principles?

So therefore: `renderRunButton` takes a boolean as input and returns a lit-html TemplateResult as output.

I think I follow this consistently for leaf-node-type of rendering problems, maybe not so much at the branch level (where the jury is out on how best to organize that code).

Reply:
yeah... having a ton of subscribers wouldn't make sense/be just as confusing.

What do you think about having the root be like suggested:
`
<div>
  <div id="code-window"></div>
  <div id="run-button"></div>
</div>
`

And then the child (which may or maynot be a leaf), `renderRunButton` takes an html element to attach to, and a boolean. Then it's stateless, but you can also read the roots to get an idea of the structure.

Reply:
and in the event the child is not a leaf, it would follow a similar pattern. Basically the parent is responsible for constructing the container that its children run in.

Reply:
Yeah but like if `renderRunButton` CAN just return a `TemplateResult`, shouldn't it?

I think one of the problems here is that the editor module is kind of a special case because of the CodeMirror situation.

Anyway let's discuss in 12h.